lib/Jenkins/API.pm

```diff
@@ -448,7 +448,7 @@ sub project_config
     my $extra_params = shift;
 
     my $uri = URI->new($self->base_url);
-    $uri->path_segments('job', $job, 'config.xml');
+    $uri->path_segments($uri->path_segments, 'job', $job, 'config.xml');
```

Comment:
That's an interesting change.  I wonder if that needs to be applied in other places too?

Reply:
It's possible, I only changed what we are using at the moment. If you'll accept this as a solution, I can go through all of `$self->base_url` places and check if it's needed or not and update them the same way as above.

Reply:
I've merged your changes.  Do you want to add yourself to the CONTRIBUTORS section of the README?

I am wondering if this is the best solution, or just a quick hack.  I am generally happy to extend the module to support doing stuff with folders, but I am wondering if it might make sense to do something different.

Your trick is interesting.  It's an option.  Another might be to have a 'jenkins_folder' method that allows you to adjust what folders the module affects.

Or all the methods could be adjusted to take extra params to allow a folder to be specified.

Reply:
Yes, this may be a quick hack for the folders, but a solution if someone is running Jenkins behind reverse-proxy with a prefix-path.

Another option would be to support project names with "/" meaning folder(s)+project_name "folder1/folder2/project_name" translating then to correct url.

Which ever option you choose, I can try help with the implementation and test with our current Jenkins set-up.

Reply:
I guess that could work too.

You are right about the proxy path thing too.  It feels like it would be good to resolve both problems.  Apply the base url regardless, and allow for / in job names, and do the right thing.